<?xml version="1.0" encoding="utf-8"?>
<robot name="MinimalRobot">
  <link name="world"/>
  <joint name="base_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
    <limit effort="0.1" lower="-3.14159265359" upper="3.14159265359" velocity="0.2"/>
  </joint>
  <link name="base_link">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="1" radius="0.1"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint1" type="revolute">
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0.2"/>
    <parent link="base_link"/>
    <child link="link1"/>
    <limit effort="0.1" lower="-3.14159265359" upper="3.14159265359" velocity="0.2"/>
  </joint>
  <link name="link1">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="1" radius="0.1"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint2" type="revolute">
    <origin rpy="1.57079632679 0 0" xyz="0 0 0.9"/>
    <parent link="link1"/>
    <child link="link2"/>
    <limit effort="0.1" lower="-3.14159265359" upper="3.14159265359" velocity="0.2"/>
  </joint>
  <link name="link2">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="1" radius="0.1"/>
      </geometry>
    </collision>
  </link>
  <ros2_control name="MockComponent" type="system">
    <hardware>
      <plugin>mock_components/GenericSystem</plugin>
    </hardware>
    <joint name="base_joint">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint1">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <joint name="joint2">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>

    <sensor name="base_ft_sensor">
      <state_interface name="force.x"/>
      <state_interface name="force.y"/>
      <state_interface name="force.z"/>
      <state_interface name="torque.x"/>
      <state_interface name="torque.y"/>
      <state_interface name="torque.z"/>
    </sensor>

    <gpio name="gpio">
      <state_interface name="digital_output_0"/>
      <state_interface name="digital_output_1"/>
      <state_interface name="digital_output_2"/>
      <state_interface name="digital_output_3"/>
      <state_interface name="digital_output_4"/>
      <state_interface name="digital_output_5"/>
      <state_interface name="digital_output_6"/>
      <state_interface name="digital_output_7"/>
      <state_interface name="digital_output_8"/>
      <state_interface name="digital_output_9"/>
      <state_interface name="digital_output_10"/>
      <state_interface name="digital_output_11"/>
      <state_interface name="digital_output_12"/>
      <state_interface name="digital_output_13"/>
      <state_interface name="digital_output_14"/>
      <state_interface name="digital_output_15"/>
    </gpio>
  </ros2_control>
</robot>
